docker/Makefile

```diff
@@ -108,3 +83,31 @@ deploy: push
 .PHONY: clean
 clean:
 	rm -f base-stmp hail-ubuntu-stmp
+
+## Public Images ###############################################################
+
+GCR_PREFIX = gcr.io/$(PROJECT)
+HAIL_IMAGE = hailgenetics/hail
+GENETICS_IMAGE = hailgenetics/genetics
+
+.PHONY: publish-hail-public-image
+publish-hail-public-image: hail-ubuntu-stmp
+	make -C ../hail python/hail/hail_pip_version
+	cp ../hail/python/hail/hail_pip_version hail/hail_pip_version
+	-docker pull $(GCR_PREFIX)/$(HAIL_IMAGE)
+	python3 ../ci/jinja2_render.py '{"hail_ubuntu_image":{"image":"hail-ubuntu"}}' Dockerfile.base Dockerfile.base.out
+	docker build hail/ -f hail/Dockerfile.out -t $(HAIL_IMAGE):$$(cat hail/hail_pip_version) \
```

Comment:
Can we use more variables here? You repeat `$$(cat hail/hail_pip_version)` a lot and it makes it harder to read. Maybe also a variable for `$(HAIL_IMAGE):$$(cat hail/hail_pip_version)`

Reply:
It's difficult to do what you're looking for in Make, so I rewrote this as a shell script. Let me know what you think.